{"create_time":"2025-04-03T19:50:00.968Z","id":"merged_46bbbe75-acc7-48f5-83b5-b7830b7ea892","source_url":"file:///C:/Users/Dj/.rustup/toolchains/1.20-x86_64-pc-windows-msvc/share/doc/rust/html/std/io/enum.ErrorKind.html","merged_triples":[{"triples":[["ErrorKind","is a","enum"],["ErrorKind","has full name","std::io::ErrorKind"],["ErrorKind","stable since","1.0.0"],["ErrorKind","has description","A list specifying general categories of I/O error."],["ErrorKind","has variant","NotFound"],["NotFound","has description","An entity was not found, often a file."],["ErrorKind","has variant","PermissionDenied"],["PermissionDenied","has description","The operation lacked the necessary privileges to complete."],["ErrorKind","has variant","ConnectionRefused"],["ConnectionRefused","has description","The connection was refused by the remote server."],["ErrorKind","has variant","ConnectionReset"],["ConnectionReset","has description","The connection was reset by the remote server."],["ErrorKind","has variant","ConnectionAborted"],["ConnectionAborted","has description","The connection was aborted (terminated) by the remote server."],["ErrorKind","has variant","NotConnected"],["NotConnected","has description","The network operation failed because it was not connected yet."],["ErrorKind","has variant","AddrInUse"],["AddrInUse","has description","A socket address could not be bound because the address is already in use elsewhere."],["ErrorKind","has variant","AddrNotAvailable"],["AddrNotAvailable","has description","A nonexistent interface was requested or the requested address was not local."],["ErrorKind","has variant","BrokenPipe"],["BrokenPipe","has description","The operation failed because a pipe was closed."],["ErrorKind","has variant","AlreadyExists"],["AlreadyExists","has description","An entity already exists, often a file."],["ErrorKind","has variant","WouldBlock"],["WouldBlock","has description","The operation needs to block to complete, but the blocking operation was requested to not occur."],["ErrorKind","has variant","InvalidInput"],["InvalidInput","has description","A parameter was incorrect."],["ErrorKind","has variant","InvalidData"],["InvalidData","has description","Data not valid for the operation were encountered. Unlike InvalidInput, this typically means that the operation parameters were valid, however the error was caused by malformed input data."],["InvalidData","stable since","1.2.0"],["ErrorKind","has variant","TimedOut"],["TimedOut","has description","The I/O operation's timeout expired, causing it to be canceled."],["ErrorKind","has variant","WriteZero"],["WriteZero","has description","An error returned when an operation could not be completed because a call to write returned Ok(0)."],["ErrorKind","has variant","Interrupted"],["Interrupted","has description","This operation was interrupted. Interrupted operations can typically be retried."],["ErrorKind","has variant","Other"],["Other","has description","Any I/O error not part of this list."],["ErrorKind","has variant","UnexpectedEof"],["UnexpectedEof","has description","An error returned when an operation could not be completed because an \"end of file\" was reached prematurely."],["UnexpectedEof","stable since","1.6.0"],["ErrorKind","implements","Debug"],["ErrorKind","implements","Clone"],["ErrorKind","implements","Eq"],["ErrorKind","implements","PartialEq"],["ErrorKind","implements","Copy"],["ErrorKind::fmt","is a","method"],["ErrorKind::fmt","has argument","self"],["ErrorKind::fmt","has argument","__arg_0"],["ErrorKind::fmt","has return type","Result"],["ErrorKind::fmt","has description","Formats the value using the given formatter."],["ErrorKind::clone","is a","method"],["ErrorKind::clone","has return type","ErrorKind"],["ErrorKind::clone","has description","Returns a copy of the value."],["ErrorKind","belongs to module","std::io"],["ErrorKind","has description","A list specifying general categories of I/O error. This list is intended to grow over time and it is not recommended to exhaustively match against it."],["InvalidData","has description","Data not valid for the operation were encountered. Unlike InvalidInput, this typically means that the operation parameters were valid, however the error was caused by malformed input data. For example, a function that reads a file into a string will error with InvalidData if the file's contents are not valid UTF-8."],["WriteZero","has description","An error returned when an operation could not be completed because a call to write returned Ok(0). This typically means that an operation could only succeed if it wrote a particular number of bytes but only a smaller number of bytes could be written."],["UnexpectedEof","has description","An error returned when an operation could not be completed because an \"end of file\" was reached prematurely. This typically means that an operation could only succeed if it read a particular number of bytes but only a smaller number of bytes could be read."],["std::io::ErrorKind","is a","enum"],["std::io::ErrorKind","stable since","1.0.0"],["std::io::ErrorKind","has description","A list specifying general categories of I/O error."],["std::io::ErrorKind","has variant","NotFound"],["std::io::ErrorKind","has variant","PermissionDenied"],["std::io::ErrorKind","has variant","ConnectionRefused"],["std::io::ErrorKind","has variant","ConnectionReset"],["std::io::ErrorKind","has variant","ConnectionAborted"],["std::io::ErrorKind","has variant","NotConnected"],["std::io::ErrorKind","has variant","AddrInUse"],["std::io::ErrorKind","has variant","AddrNotAvailable"],["std::io::ErrorKind","has variant","BrokenPipe"],["std::io::ErrorKind","has variant","AlreadyExists"],["std::io::ErrorKind","has variant","WouldBlock"],["std::io::ErrorKind","has variant","InvalidInput"],["std::io::ErrorKind","has variant","InvalidData"],["std::io::ErrorKind","has variant","TimedOut"],["std::io::ErrorKind","has variant","WriteZero"],["std::io::ErrorKind","has variant","Interrupted"],["std::io::ErrorKind","has variant","Other"],["std::io::ErrorKind","has variant","UnexpectedEof"],["ErrorKind::NotFound","has description","An entity was not found, often a file."],["ErrorKind::PermissionDenied","has description","The operation lacked the necessary privileges to complete."],["ErrorKind::ConnectionRefused","has description","The connection was refused by the remote server."],["ErrorKind::ConnectionReset","has description","The connection was reset by the remote server."],["ErrorKind::ConnectionAborted","has description","The connection was aborted (terminated) by the remote server."],["ErrorKind::NotConnected","has description","The network operation failed because it was not connected yet."],["ErrorKind::AddrInUse","has description","A socket address could not be bound because the address is already in use elsewhere."],["ErrorKind::AddrNotAvailable","has description","A nonexistent interface was requested or the requested address was not local."],["ErrorKind::BrokenPipe","has description","The operation failed because a pipe was closed."],["ErrorKind::AlreadyExists","has description","An entity already exists, often a file."],["ErrorKind::WouldBlock","has description","The operation needs to block to complete, but the blocking operation was requested to not occur."],["ErrorKind::InvalidInput","has description","A parameter was incorrect."],["ErrorKind::InvalidData","has description","Data not valid for the operation were encountered."],["ErrorKind::InvalidData","stable since","1.2.0"],["ErrorKind::TimedOut","has description","The I/O operation's timeout expired, causing it to be canceled."],["ErrorKind::WriteZero","has description","An error returned when an operation could not be completed because a call to write returned Ok(0)."],["ErrorKind::Interrupted","has description","This operation was interrupted."],["ErrorKind::Other","has description","Any I/O error not part of this list."],["ErrorKind::UnexpectedEof","has description","An error returned when an operation could not be completed because an \"end of file\" was reached prematurely."],["ErrorKind::UnexpectedEof","stable since","1.6.0"],["std::io::ErrorKind","implements","Debug"],["std::io::ErrorKind","implements","Clone"],["std::io::ErrorKind","implements","Eq"],["std::io::ErrorKind","implements","PartialEq"],["std::io::ErrorKind","implements","Copy"],["ErrorKind::fmt","has return type","std::fmt::Result"],["ErrorKind","used with","io::Error"],["ErrorKind","implements","PartialOrd"],["std::io::ErrorKind","implements","PartialOrd"],["ErrorKind::partial_cmp","is a","method"],["ErrorKind::partial_cmp","has argument","self"],["ErrorKind::partial_cmp","has argument","__arg_0"],["ErrorKind::partial_cmp","has return type","Option<Ordering>"],["ErrorKind::partial_cmp","has description","This method returns an ordering between self and other values if one exists."],["ErrorKind::lt","is a","method"],["ErrorKind::lt","has argument","self"],["ErrorKind::lt","has argument","other"],["ErrorKind::lt","has return type","bool"],["ErrorKind::lt","has description","This method tests less than (for self and other) and is used by the < operator."],["ErrorKind::le","is a","method"],["ErrorKind::le","has argument","self"],["ErrorKind::le","has argument","other"],["ErrorKind::le","has return type","bool"],["ErrorKind::le","has description","This method tests less than or equal to (for self and other) and is used by the <= operator."],["ErrorKind::gt","is a","method"],["ErrorKind::gt","has argument","self"],["ErrorKind::gt","has argument","other"],["ErrorKind::gt","has return type","bool"],["ErrorKind::gt","has description","This method tests greater than (for self and other) and is used by the > operator."],["ErrorKind::ge","is a","method"],["ErrorKind::ge","has argument","self"],["ErrorKind::ge","has argument","other"],["ErrorKind::ge","has return type","bool"],["ErrorKind::ge","has description","This method tests greater than or equal to (for self and other) and is used by the >= operator."],["std::io::ErrorKind","has description","A list specifying general categories of I/O error. This list is intended to grow over time and it is not recommended to exhaustively match against it."],["ErrorKind::InvalidData","has description","Data not valid for the operation were encountered. Unlike InvalidInput, this typically means that the operation parameters were valid, however the error was caused by malformed input data. For example, a function that reads a file into a string will error with InvalidData if the file's contents are not valid UTF-8."],["ErrorKind::WriteZero","has description","An error returned when an operation could not be completed because a call to write returned Ok(0). This typically means that an operation could only succeed if it wrote a particular number of bytes but only a smaller number of bytes could be written."],["ErrorKind::UnexpectedEof","has description","An error returned when an operation could not be completed because an \"end of file\" was reached prematurely. This typically means that an operation could only succeed if it read a particular number of bytes but only a smaller number of bytes could be read."],["std::io::ErrorKind","implements","Hash"],["std::io::ErrorKind::partial_cmp","is a","method"],["std::io::ErrorKind::partial_cmp","has argument","self"],["std::io::ErrorKind::partial_cmp","has argument","__arg_0"],["std::io::ErrorKind::partial_cmp","has return type","Option<Ordering>"],["std::io::ErrorKind::partial_cmp","has description","This method returns an ordering between self and other values if one exists."],["std::io::ErrorKind::lt","is a","method"],["std::io::ErrorKind::lt","has argument","self"],["std::io::ErrorKind::lt","has argument","other"],["std::io::ErrorKind::lt","has return type","bool"],["std::io::ErrorKind::lt","has description","This method tests less than (for self and other) and is used by the < operator."],["std::io::ErrorKind::le","is a","method"],["std::io::ErrorKind::le","has argument","self"],["std::io::ErrorKind::le","has argument","other"],["std::io::ErrorKind::le","has return type","bool"],["std::io::ErrorKind::le","has description","This method tests less than or equal to (for self and other) and is used by the <= operator."],["std::io::ErrorKind::gt","is a","method"],["std::io::ErrorKind::gt","has argument","self"],["std::io::ErrorKind::gt","has argument","other"],["std::io::ErrorKind::gt","has return type","bool"],["std::io::ErrorKind::gt","has description","This method tests greater than (for self and other) and is used by the > operator."],["std::io::ErrorKind::ge","is a","method"],["std::io::ErrorKind::ge","has argument","self"],["std::io::ErrorKind::ge","has argument","other"],["std::io::ErrorKind::ge","has return type","bool"],["std::io::ErrorKind::ge","has description","This method tests greater than or equal to (for self and other) and is used by the >= operator."]],"startIndex":0,"endIndex":39000}],"changes_info_triples":[["ErrorKind","adds feature","PartialOrd"],["std::io::ErrorKind","adds feature","PartialOrd"],["ErrorKind::fmt","changes return type to","std::fmt::Result"],["ErrorKind","removes feature","Copy"],["std::io::ErrorKind","removes feature","Copy"],["ErrorKind","adds usage with","io::Error"],["ErrorKind","adds method","partial_cmp"],["ErrorKind","adds method","lt"],["ErrorKind","adds method","le"],["ErrorKind","adds method","gt"],["ErrorKind","adds method","ge"],["ErrorKind::InvalidData","removes description","Data not valid for the operation were encountered."],["ErrorKind","updates description","A list specifying general categories of I/O error. This list is intended to grow over time and it is not recommended to exhaustively match against it."],["InvalidData","updates description","Data not valid for the operation were encountered. Unlike InvalidInput, this typically means that the operation parameters were valid, however the error was caused by malformed input data. For example, a function that reads a file into a string will error with InvalidData if the file's contents are not valid UTF-8."],["WriteZero","updates description","An error returned when an operation could not be completed because a call to write returned Ok(0). This typically means that an operation could only succeed if it wrote a particular number of bytes but only a smaller number of bytes could be written."],["UnexpectedEof","updates description","An error returned when an operation could not be completed because an \"end of file\" was reached prematurely. This typically means that an operation could only succeed if it read a particular number of bytes but only a smaller number of bytes could be read."],["ErrorKind","adds method","fmt"],["ErrorKind","adds method","clone"],["std::io::ErrorKind","updates description","A list specifying general categories of I/O error. This list is intended to grow over time and it is not recommended to exhaustively match against it."],["ErrorKind::InvalidData","updates description","Data not valid for the operation were encountered. Unlike InvalidInput, this typically means that the operation parameters were valid, however the error was caused by malformed input data. For example, a function that reads a file into a string will error with InvalidData if the file's contents are not valid UTF-8."],["ErrorKind::WriteZero","updates description","An error returned when an operation could not be completed because a call to write returned Ok(0). This typically means that an operation could only succeed if it wrote a particular number of bytes but only a smaller number of bytes could be written."],["ErrorKind::UnexpectedEof","updates description","An error returned when an operation could not be completed because an \"end of file\" was reached prematurely. This typically means that an operation could only succeed if it read a particular number of bytes but only a smaller number of bytes could be read."],["std::io::ErrorKind","adds feature","Hash"],["std::io::ErrorKind","adds method","partial_cmp"],["std::io::ErrorKind","adds method","lt"],["std::io::ErrorKind","adds method","le"],["std::io::ErrorKind","adds method","gt"],["std::io::ErrorKind","adds method","ge"]],"model":"deepseek-r1-250120"}